sound/snd_efxfile.cpp

```diff
@@ -252,6 +265,96 @@ bool idEFXFile::ReadEffectLegacy(idLexer &src, idSoundEffect *effect) {
 	return true;
 }
 
+bool idEFXFile::AddOrUpdatePreset(idStr areaName, idStr efxPreset, ALuint* effect) {
+
+	idSoundEffect* soundEffect = new idSoundEffect;
+
+	//ALuint effect = AL_EFFECTSLOT_NULL;
+	const bool found = GetEffect(areaName, soundEffect);
+	ALenum err{};
+	bool ok;
+
+	if (!found) {
+
+		// we need to add this preset
+
+		if (!soundEffect->alloc()) {
+			delete soundEffect;
+			Clear();
+			return false;
+		}
+
+		soundEffect->name = areaName;
+
+		//ALenum err{};
+		bool ok;
+		ok = AddPreset(efxPreset, soundEffect, err);
+
+		if (!ok) {
+			return false;
+		}
+
+		effects.Append(soundEffect);
+	}
+	else {
+		ok = AddPreset(efxPreset, soundEffect, err);
```

Comment:
I don't understand why both cases call AddPreset.

Reply:
Presets can be used by either the EFX file or the spawnarg, so instead of duplicating the code I refactored it into a new function.

Reply:
Sorry ignore my last comment - I understand what you mean now.  I've gone back and refactored it, it should look better now.